provisioning/device/src/Models/DeviceRegistrationResult.cs

```diff
@@ -0,0 +1,182 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client
+{
+    /// <summary>
+    /// The result of a registration operation.
+    /// </summary>
+    public class DeviceRegistrationResult
+    {
+        /// <summary>
+        /// For serialization.
+        /// </summary>
+        internal DeviceRegistrationResult()
+        {
+
+        }
+
+        /// <summary>
+        ///. Creates an instance of this class.
+        /// </summary>
+        public DeviceRegistrationResult(
+            string registrationId,
+            DateTime? createdDateTimeUtc,
+            string assignedHub,
+            string deviceId,
+            ProvisioningRegistrationStatusType status,
+            ProvisioningRegistrationSubstatusType substatus,
+            string generationId,
+            DateTime? lastUpdatedDateTimeUtc,
+            int errorCode,
+            string errorMessage,
+            string etag,
+            JRaw jsonPayload)
+        {
+            RegistrationId = registrationId;
+            CreatedDateTimeUtc = createdDateTimeUtc;
+            AssignedHub = assignedHub;
+            DeviceId = deviceId;
+            Status = status;
+            Substatus = substatus;
+            GenerationId = generationId;
+            LastUpdatedDateTimeUtc = lastUpdatedDateTimeUtc;
+            ErrorCode = errorCode;
+            ErrorMessage = errorMessage;
+            Etag = etag;
+            Payload = jsonPayload;
+        }
+
+        /// <summary>
+        ///. Creates an instance of this class.
+        /// </summary>
+        internal DeviceRegistrationResult(
+            string registrationId,
+            DateTime? createdDateTimeUtc,
+            string assignedHub,
+            string deviceId,
+            ProvisioningRegistrationStatusType status,
+            ProvisioningRegistrationSubstatusType substatus,
+            string generationId,
+            DateTime? lastUpdatedDateTimeUtc,
+            int errorCode,
+            string errorMessage,
+            string etag,
+            JRaw jsonPayload,
+            TpmRegistrationResult tpm = default,
+            X509RegistrationResult x509 = default,
+            SymmetricKeyRegistrationResult symmetricKey = default)
+            : this(
+                  registrationId,
+                  createdDateTimeUtc,
+                  assignedHub,
+                  deviceId,
+                  status,
+                  substatus,
+                  generationId,
+                  lastUpdatedDateTimeUtc,
+                  errorCode,
+                  errorMessage,
+                  etag,
+                  jsonPayload)
+        {
+            Tpm = tpm;
+            X509 = x509;
+            SymmetricKey = symmetricKey;
+        }
+
+        /// <summary>
+        /// The registration id.
+        /// </summary>
+        [JsonProperty(PropertyName = "registrationId")]
+        public string RegistrationId { get; internal set; }
```

Comment:
we probably don't need these setters anymore either

Reply:
We need them for deserialization, right?

Reply:
ah, you're right